api/users/serializers.py

```diff
@@ -128,6 +132,67 @@ def profile_image_url(self, user):
         size = self.context['request'].query_params.get('profile_image_size')
         return user.profile_image_url(size=size)
 
+    def _update_social(self, social_dict, user):
+        for key, val in social_dict.items():
+            # currently only profileWebsites are a list, the rest of the social key only has one value
+            if key == 'profileWebsites':
+                user.social[key] = val
+            else:
+                if len(val) > 1:
+                    raise InvalidModelValueError(
+                        detail='{} only accept a list of one single value'.format(key)
+                    )
+                    user.social[key] = val[0]
+
+    def _validate_user_json(self, info_list, required_fields):
```

Comment:
I recommend using `jsonschema` for most of this validation, instead of individually type checking each field.  We use this for validating form information for [draft registrations](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/metaschema.py#L73).  I doubt jsonschema can validate for everything that you want, for example, end date must be greater that start date, but it works well for required fields, field types, etc.  You can also define field dependencies (if endMonth is present, endYear must be present, for example).

You would do something like: 
- import jsonschema 
- define your schema, started below: 
- `jsonschema.validate(validated_data, schema)`

Here's a start on the schema for education data, doesn't have everything:
```json
{
  "type": "array",
  "items": {
    "type": "object",
    "properties": {
      "degree": {
        "type": "string",
      },
      "startYear": {
        "type": "integer",
      },
      "startMonth": {
        "type": "integer"
      },
      "endMonth": {
        "type": "integer"
      },
      "endYear": {
        "type": "integer"
      },
      "ongoing": {
        "type": "boolean"
      },
      "department": {
        "type": "string"
      },
      "institution": {
        "type": "string"
      }
    },
    "required": [
        "startYear",
        "startMonth",
        "ongoing",
        "department",
        "institution",
        "degree"
    ],
    "dependencies": {
        "endMonth": ["endYear"],
        "endYear": ["endMonth"]
     }
  }
}
```